isamples_metadata/OpenContextTransformer.py

```diff
@@ -0,0 +1,217 @@
+        ],
+        "Organism part",
+    )
+    _anthropogenicAggregationMapper = StringEqualityCategoryMapper(
+        ["Architectural Element", "Basket", "Bulk Ceramic", "Lot"],
+        "Anthropogenic aggregation",
+    )
+    _biomeAggregationMapper = StringEqualityCategoryMapper(
+        ["Biological subject, Ecofact", "Plant remains"], "Biome aggregation"
+    )
+    _artifactMapper = StringEqualityCategoryMapper(
+        ["Coin", "Glass", "Groundstone", "Object", "Pottery", "Sculpture", "Stela"],
+        "Artifact",
+    )
+    _otherSolidObjectMapper = StringEqualityCategoryMapper(
+        ["Sample", "Sample, Collection, or Aggregation"], "Other solid object"
+    )
+    _organismProductMapper = StringEqualityCategoryMapper(
+        [
+            "Shell",
+        ],
+        "Organism product",
+    )
+
+    @classmethod
+    def categories_mappers(cls) -> typing.List[AbstractCategoryMapper]:
+        return [
+            cls._organismPartMapper,
+            cls._anthropogenicAggregationMapper,
+            cls._biomeAggregationMapper,
+            cls._artifactMapper,
+            cls._otherSolidObjectMapper,
+            cls._organismProductMapper,
+        ]
+
+
+class OpenContextTransformer(Transformer):
+    N2T_PREFIX = "https://n2t.net/"
+    N2T_ARK_PREFIX = f"{N2T_PREFIX}ark:/"
+
+    def transform(self) -> typing.Dict:
+        transformed_record = super().transform()
+        return transformed_record
+
+    def _citation_uri(self) -> typing.AnyStr:
+        return self.source_record.get("citation uri")
+
+    def id_string(self) -> typing.AnyStr:
+        citation_uri = self._citation_uri()
+        return f"metadata/{citation_uri.removeprefix(self.N2T_ARK_PREFIX)}"
+
+    def sample_identifier_scheme(self) -> typing.AnyStr:
+        pass
+
+    def sample_identifier_string(self) -> typing.AnyStr:
+        return self._citation_uri().removeprefix(self.N2T_PREFIX)
+
+    def sample_identifier_value(self) -> typing.AnyStr:
+        # TODO is this even used?
+        return self._citation_uri().removeprefix(self.N2T_PREFIX)
+
+    def sample_label(self) -> typing.AnyStr:
+        return self.source_record.get("label", Transformer.NOT_PROVIDED)
+
+    def sample_description(self) -> typing.AnyStr:
+        description_pieces = []
+        self._transform_key_to_label(
+            "early bce/ce", self.source_record, description_pieces
+        )
+        self._transform_key_to_label(
+            "late bce/ce", self.source_record, description_pieces
+        )
+        self._transform_key_to_label("updated", self.source_record, description_pieces)
+        self._transform_key_to_label("updated", self.source_record, description_pieces)
+        for consistsOfDict in self.source_record.get("Consists of", []):
+            self._transform_key_to_label(
+                "label", consistsOfDict, description_pieces, "Consists of"
+            )
+        for hasTypeDict in self.source_record.get("Has type", []):
```

Comment:
We have some other attributes ("Has taxonomic identifier", "Has anatomical identification", etc.) that may need treatment like this, or are those handled elsewhere?

Reply:
Ahhh, I probably just missed them.  I'll take a look.  Could you point me at a couple records that have those?

Reply:
Looks like http://opencontext.org/subjects/DB96F124-4E2D-42C8-06FD-62C6DD714AB3 has these two.